thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -349,10 +355,33 @@ public void serviceAdded(ServiceConfig cfg) throws Exception {
                                 format, maxRequestStringLength, maxRequestContainerLength)));
 
         super.serviceAdded(cfg);
+
+        final DependencyInjector dependencyInjector = cfg.server().config().dependencyInjector();
+        for (ThriftServiceEntry thriftServiceEntry : entries().values()) {
+            for (ThriftFunction thriftFunction : thriftServiceEntry.metadata.functions().values()) {
+                if (!thriftFunction.declaredDecorators().isEmpty()) {
+                    final List<DecoratorAndOrder> decorators = thriftFunction.declaredDecorators();
+                    Function<? super HttpService, ? extends HttpService> decorator = Function.identity();
+                    for (int i = decorators.size() - 1; i >= 0; i--) {
+                        final DecoratorAndOrder d = decorators.get(i);
+                        decorator = decorator.andThen(d.decorator(dependencyInjector));
+                    }
+                    thriftFunction.setDecoratedTHttpService(decorator.apply(this));
```

Comment:
If possible, an immutable data structure is preferred. What do you think of creating a map whose key is `ThriftFunction` and whose value is a decorated service instead of mutating `thriftFunction`? For example:
```java
private Map<ThriftFunction, HttpService> decorators;
...
decorators.put(thriftFunction, decorator.apply(this));
```

Reply:
I agree :)
So I tried to apply your suggestion. PTAL 🙇🏻‍♂️